apps/api/v2/src/modules/availabilities/inputs/create-availability.input.ts

```diff
@@ -19,12 +19,17 @@ export class CreateAvailabilityInput {
 }
 
 function transformStringToDate(value: string, key: string): Date {
-  const parts = value.split(":");
+  // note(Lauris): incoming value is ISO8061 e.g. 2025-0412T13:17:56.324Z
+  const dateTimeParts = value.split("T");
+
+  const timePart = dateTimeParts[1].split(".")[0]; // Removes milliseconds
+  const parts = timePart.split(":");
 
   if (parts.length !== 3) {
-    throw new BadRequestException(`Invalid ${key} format. Expected format: HH:MM:SS. Received ${value}`);
+    throw new BadRequestException(
+      `Invalid time format. Expected format(ISO8061): 2025-0412T13:17:56.324Z. Received: ${value}`
```

Comment:
Feels like for availabilities that are using on the 0 second we should return back something like that.

e.g. `2025-0412T13:00:00Z` or `2025-0412T13:30:00Z`

Reply:
I'm not even sure our system can support saving availabilities with times like these

Reply:
I checked payload of when availability is updated and "Save" is hit:

<img width="1482" alt="Screenshot 2024-04-12 at 17 28 11" src="https://github.com/calcom/cal.com/assets/42170848/84331c16-a0af-4907-a465-523322563766">

And then I matched behaviour exactly in the code above. There are more elegant ways to write this input DTO but need to think about it and potentially refactor how the availability atom works.

Reply:
Yeah that ISO format is definitely the one we want to support but what I was referring to is the date itself. The UI doesn’t support setting your availability down to the second.